chapter03/pxd/ex2.pl

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/perl -w
+
+use strict;
+use feature "say";
+
+my $dir = "/usr/lib/perl5/DataCash/*";
+
+#opendir(D, "$dir") || die "Can't open directory $dir: $!\n";
+#my @files = readdir($dir);
+#closedir($dir);
+#my @files = glob $dir;
+
+my @files = </usr/lib/perl5/DataCash/*>;
+
+my $user_regex;
+
+if ( scalar @ARGV > 0 ) {
+  say "You can't do that!";
+  exit 1;
+}
+
+say "Please enter a regex to match or type quit to exit: ";
+
+while ( chomp (  $user_regex = <STDIN> ) ) {
+
+  if ( !$user_regex || ($user_regex =~ /quit/  )) {
+	say "Please enter a regex to proceed...";
+	next;
```

Comment:
True, I had initially just added the first check and extended it to quit and forgot about the `next` bit.